modules/assim.batch/R/pda.emulator.ms.R

```diff
@@ -0,0 +1,518 @@
+##' Paramater Data Assimilation using emulator on multiple sites in three modes: local, global, hierarchical
+##' First draft, not complete yet
+##'
+##' @param multi.settings = a pecan multi-settings list
+##'
+##' @return settings
+##'
+##' @author Istem Fer
+##' @export
+pda.emulator.ms <- function(multi.settings) {
+  
+  ## -------------------------------------- Initialization --------------------------------------------------- 
+ 
+  # check mode 
+  pda.mode <- unique(sapply(multi.settings$assim.batch,`[[`, "mode"))
+  
+  if(pda.mode == "local"){
+    local <- TRUE
+    global <- hierarchical <- FALSE
+  }else if(pda.mode == "global"){
+    global <- TRUE
+    local <- hierarchical <- FALSE
+  }else if(pda.mode == "hierarchical"){
+    hieararchical <- TRUE
+    local <- global <- FALSE
+  }else{
+    local <- global <- hierarchical <- TRUE
+  }
+  
+  # how many sites
+  nsites <- length(multi.settings)
+
+  
+  # lists to collect emulators and run MCMC per site later
+  gp.stack      <- vector("list", nsites) 
+  SS.stack      <- vector("list", nsites) 
+  #nstack       <- vector("list", nsites) 
+  
+  ## -------------------------------------- Local runs and calibration ------------------------------------------
+  
+  for(s in seq_along(multi.settings)){ # site runs - loop begin
```

Comment:
yeah I'll change these to papply, but I want to note that it's a little too tolerant for failures. I only realized after three sites ran that I had a typo in the xml that would normally cause error in the first pass otherwise

>Speaking of parallelization, if pda.emulator does not return until it's done then we're really missing the ability to leverage remote by putting all runs for all sites into the queue at once.

This is true, I will look into this later.

>Also, I assume pda.emulator can be configured to do multiple rounds of knots per site automatically?

It doesn't do it automatically yet. If user decides they need another round they need to re-read the xml written by the pda-workflow and re-start pda.emulator. But automation shouldn't be difficult.